hexa/databases/schema.py

```diff
@@ -38,6 +40,37 @@ def resolve_database_table(workspace, info, **kwargs):
     return get_table_definition(workspace, kwargs.get("name"))
 
 
+@database_object.field("port")
+def resolve_database_port(_, info, **kwargs):
+    return get_db_server_credentials()["port"]
+
+
+@database_object.field("host")
+def resolve_database_host(_, info, **kwargs):
+    return get_db_server_credentials()["host"]
+
+
+@database_object.field("password")
+def resolve_database_password(workspace, info, **kwargs):
```

Comment:
In the end, we will have a `resolve_database_credentials` resolver. 

It should call `has_perm("database.view_database_credentials")`. And in this new permission function, you should check whether the user is either admin or editor of the workspace.